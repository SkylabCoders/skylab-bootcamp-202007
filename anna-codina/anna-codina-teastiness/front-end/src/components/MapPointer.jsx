import React from 'react';
import { Link } from 'react-router-dom';
import BodyBoxShop from './BodyBoxShop';

const MapPointer = ({ shop }) => {
  return (
    <Link className="map-pointer" to={`/shop/${shop._id}`}>
      <img
        className="map-pointer__token"
        alt="pointer"
        src="https://image.flaticon.com/icons/svg/3043/3043769.svg"
      ></img>
      <BodyBoxShop shop={shop} />
    </Link>
  );
};
export default MapPointer;
