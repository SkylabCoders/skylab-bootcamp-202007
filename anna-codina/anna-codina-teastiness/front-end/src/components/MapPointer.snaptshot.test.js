import renderer from 'react-test-renderer';
import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import MapPointer from './MapPointer';

function renderMapPointer(shop) {
  return renderer.create(
    <Router>
      <MapPointer shop={shop} />
    </Router>
  );
}

describe('App snapshot', () => {
  let mapPointerTree;
  const shop = { _id: 13 };
  beforeEach(async () => {
    mapPointerTree = renderMapPointer(shop);
    mapPointerTree.update();
  });

  it('should match', async () => {
    mapPointerTree = renderMapPointer(shop);
    expect(mapPointerTree).toMatchSnapshot();
  });
});
