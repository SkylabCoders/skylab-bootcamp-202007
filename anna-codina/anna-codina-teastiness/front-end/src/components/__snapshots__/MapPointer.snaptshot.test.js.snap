// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot should match 1`] = `
<a
  className="map-pointer"
  href="/shop/13"
  onClick={[Function]}
>
  <img
    alt="pointer"
    className="map-pointer__token"
    src="https://image.flaticon.com/icons/svg/3043/3043769.svg"
  />
</a>
`;
